src/build/env.py

```diff
@@ -215,27 +259,74 @@ def __enter__(self):  # type: () -> IsolatedEnvironment
 
         The environment path should be empty
         """
-        self._path = tempfile.mkdtemp(prefix='build-env-')
+        if self._path is None:
+            self._path = tempfile.mkdtemp(prefix='build-env-')
         self._env_vars = {
             'base': self.path,
             'platbase': self.path,
         }
 
+        with self._elapsed('Done isolated build environment at {} '.format(self.path)):
+            self._create_isolated_python()
+            self._provision_pip()
+
+        sys.executable, self._old_sys_executable = self._executable, sys.executable
+        self._pop_env('PIP_REQUIRE_VIRTUALENV')
+
+        return self
+
+    @contextmanager
+    def _elapsed(self, msg):
+        start = datetime.now()
+        try:
+            yield
+        finally:
+            print('{} within {}'.format(msg, (datetime.now() - start).total_seconds()))
+
+    def _create_isolated_python(self):
         if sys.version_info[0] == 2:
             self._create_env_pythonhome()
         else:
             self._create_env_venv()
 
-        self._pop_env('PIP_REQUIRE_VIRTUALENV')
-
-        return self
+    def _provision_pip(self):
+        if self._isolation.ensure_pip:
+            subprocess.check_call([self._executable, '-Im', 'ensurepip', '--upgrade', '--default-pip'])
+            # we don't want the default setuptools
+            subprocess.check_call([self._executable, '-m', 'pip', 'uninstall', 'setuptools', '-y'])
+        else:
+            # version qualify per python minor, so that we don't break when pip drops older python versions
+            folder = os.path.expanduser(os.path.expandvars(self._isolation.cache))
+            if self._isolation.reset_cache and os.path.exists(folder):
+                print('Delete {}'.format(folder))
```

Comment:
Remove this print. We could add a logging system if needed, but we should not be using print IMO.

Reply:
I mean the most popular backend setuptools already uses print, and we don't interrupt that. So there's already tons of print in the code. So not sure I follow on your worry.